File: src/domain/entities/Course.ts
import { SubjectType } from '@/core/types/common';

export interface Course {
  id: string;
  title: string;
  subject: SubjectType;
  description: string;
  iconName: string;
  primaryColor: string;
  secondaryColor: string;
  skills: string[]; // skill IDs
  lessons: string[]; // lesson IDs
  totalStudentsEstimate?: number;
}

File: mobile/lib/domain/entities/course.dart
enum SubjectType {
  mathematics,
  english,
}

class Course {
  final String id;
  final String title;
  final SubjectType subject;
  final String description;
  final String iconName;
  final String primaryColorHex;
  final String secondaryColorHex;
  final List<String> skills;
  final List<String> lessons;
  final int totalStudentsEstimate;

  const Course({
    required this.id,
    required this.title,
    required this.subject,
    required this.description,
    required this.iconName,
    required this.primaryColorHex,
    required this.secondaryColorHex,
    required this.skills,
    required this.lessons,
    this.totalStudentsEstimate = 1000,
  });
}
